src/libecalc/application/energy_calculator.py

```diff
@@ -49,14 +48,14 @@ def __init__(
     ):
         self._graph = graph
 
-    def evaluate_energy_usage(self, variables_map: VariablesMap) -> Dict[str, EcalcModelResult]:
+    def evaluate_energy_usage(self, variables_map: VariablesMap) -> dict[str, EcalcModelResult]:
         component_ids = list(reversed(self._graph.sorted_node_ids))
         component_dtos = [self._graph.get_node(component_id) for component_id in component_ids]
 
-        consumer_results: Dict[str, EcalcModelResult] = {}
+        consumer_results: dict[str, EcalcModelResult] = {}
 
         for component_dto in component_dtos:
-            if isinstance(component_dto, (ElectricityConsumerDTO, FuelConsumerDTO)):
+            if isinstance(component_dto, ElectricityConsumerDTO | FuelConsumerDTO):
```

Comment:
Had to check this. Unions can now be written with `|`, which wasn't new to me, but `isinstance` now supports unions since 3.10. So `isinstance(component_dto, Union[ElectricityConsumerDTO, FuelConsumerDTO])` would also work, I guess. `|` is still a type-thing..

Reply:
aha, was just following rust-recommendations. We can (and should?) update Unions to | , but it was not forced, so skipped it, for simplicity ...